added code and test to export darwin dataset as semantic segmentation

BartvanMarrewijk opened the pull request:
Added code to export a dataset for semantic segmentation.

- added as_coco_semseg to create semantic segmentation images and json.
- Both json and images are stored in annotation_path!
- added feature when segmentation_order is important
- added option to skip classes, we deliberately chose to no filter the annotations beforehands, but after the mask is created (broccoli usecase)
- fixed issue in darwin.py when you have items with properties

BartvanMarrewijk commented:
@roel-klein

roel-klein commented on supervision/dataset/core.py:
I'm not sure what segmentation order does now, please add some more description

roel-klein commented on supervision/dataset/formats/coco.py:
```diff
@@ -265,6 +266,156 @@ def save_coco_annotations(
+        ## skip classes if exist
+        if class_id in idx_skip_classes:
+            continue
+        mask = np.maximum(mask, (instance_mask.astype(np.uint8) * (class_id)))
```
why maximum, does this mean that classes with higher values end up on top?

BartvanMarrewijk replied:
good point fixed using np.where

roel-klein commented on supervision/dataset/formats/coco.py:
```diff
@@ -265,6 +266,156 @@ def save_coco_annotations(
+        )
+        image_name = str(image_path_relative)
+
+        ## if true a semantic mask will be created per box
```
please split it up a bit

BartvanMarrewijk commented:
@roel-klein 

big update with adding semantic segmentation as separate format. important changes:
in **dataset core:**
from supervision.dataset.formats.coco_semseg import (
    load_coco_semseg_annotations,
    load_from_semseg_dir,
    save_coco_semseg_annotations,
)

Added **datset/formats/coco_semseg.py**

moved find_valid_images_and_annotations to supervision.utils.file. This will likely break mlutils but lets fix that quickly after this merge

tools/darwin.py -> **bug fix for list items in darwin annotations**
detection/utils.py -> masks_to_semantic_mask function
supervision/utils/image.py --> added advanced cropping based on xyxy
